### PR TITLE
Fragments

### PR DESCRIPTION
**Pull request by abea**
It's still written in the migratory perspective, but hopefully in such a way as to be easily updated for new devs later. Ultimately we'll separate "widgets and templates," but that was out of scope for this mission.

### CODE REVIEW

File: docs/guide/widgets-and-templates/fragments.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+{% fragment heading() %}
+  <h2>This is a heading fragment</h2>
+{% endfragment %}
+```
+
+One difference from macros is that they are then called using the `render` template tag:
+
+```django
+{% render heading() %}
+{# `<h2>This is a heading fragment</h2>` #}
```
Comment (boutell) [inline docs/guide/widgets-and-templates/fragments.md:31]: I'm not sure the comment syntax is a good way to express this, maybe a separate markup code block showing the output.

File: docs/guide/widgets-and-templates/fragments.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+You can also pass arguments to macros, which is key to using them across templates. Those arguments can be any data or variables understood by Apostrophe templates (e.g., strings, objects, arrays, etc.).
```
Comment (boutell) [inline docs/guide/widgets-and-templates/fragments.md:34]: To fragments.

File: docs/guide/widgets-and-templates/fragments.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+```
+
+## Fragment limitations
+
+In comparing fragments to macros, it is worth noting a couple of macro features that are not *yet* supported in fragments.
+
+
+### Default parameter values
+
+Macros can include default values for their arguments. This is not currently available in fragments. As a work-around, you can set variables in fragments that are overwritten if the argument is provided.
```
Comment (boutell) [inline docs/guide/widgets-and-templates/fragments.md:119]: Nice

**Review by boutell — CHANGES_REQUESTED** [review]

File: docs/guide/widgets-and-templates/fragments.md (hunk 1 above)
Comment (abea) [reply docs/guide/widgets-and-templates/fragments.md:31]: Improved to `{# Renders: `<h2>This is a heading fragment</h2>` #}`?

File: docs/guide/widgets-and-templates/home-page.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+// modules/@apostrophecms/home-page/views/page.html
+{% extends "layout.html" %}
+
+{% block main %}
+  {% area data.page, 'main' %}
+{% endblock %}
+```
+
+- Areas are added to the template with the new `area` nunjucks tag. There is no `apos.area` helper function anymore.
+- You no longer configure the area here. This is done exclusively in the `index.js` file for the page-type or piece-type. In the template, just pass the page and the name of the area.
```
Comment (boutell) [inline docs/guide/widgets-and-templates/home-page.md:15]: Which means this example is not sufficient, so an example index.js should to be given too, declaring the main area as a field.

**Review by boutell — CHANGES_REQUESTED** [review]
Just one change, re: home pages.

File: docs/guide/widgets-and-templates/home-page.md (hunk 4 above)
Comment (abea) [reply docs/guide/widgets-and-templates/home-page.md:15]: This was transferred from the `widgets-and-templates.md` file verbatim. The whole page needs to be rewritten or replaced, but I don't think this is the place to do it. Configuring areas is covered elsewhere as well.

**Review by boutell — APPROVED** [review]

**Pull request merged**